src/commands/deploy/deploy.ts

```diff
@@ -158,7 +158,8 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
         deployContext.telemetry.properties.cancelStep = '';
     }
 
-    if (!deployContext.deployedWithConfigs) {
+    const defaultWebAppToDeploySetting: string | undefined = getWorkspaceSetting(constants.configurationSettings.defaultWebAppToDeploy, deployFsPath);
```

Comment:
> I wanted to change deployWithConfigs to include a none type, but getDefaultWebApp doesn't get called if the user entered via the context menu (since we already have a node) so I have to check the workspace setting instead.

We usually have a separate setting like `appService.showDefaultWebAppPrompt` that is checked regardless of the entry point. Instead, the `none` value essentially mashes the `defaultWebAppToDeploy` setting with `showDefaultWebAppPrompt` which is why you ran into that problem. That being said I wouldn't want to change this without discussing backwards compatibility.

Reply:
I could write migration code that checks for the `none` value in showDefaultWebApp prompt and then sets `appService.showDefaultWebAppPrompt`.  That seems simple enough, but is an ugly solution that would have to remain in the codebase _FOREVER_ 😱

Reply:
It wouldn't have to be there _forever_. We could deprecate the old setting and remove it eventually